From 1e57b5c4ba1812807a90c7b15bcacb9e10fe9ef5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ant=C3=B3nio=20Afonso?= <aadsm@users.noreply.github.com>
Date: Wed, 9 Feb 2022 12:41:08 -0800
Subject: [PATCH] Don't use `parser` module if `symbol` doesn't exist as well.

Python 3.10 removed the `parser` and the `symbol` modules that were deprecated for a while now.
There's a fallback implementation with markerlib for `parser` but we were not checking for the lack of the `symbol` module as well.
---
 third-party/py/setuptools/pkg_resources/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third-party/py/setuptools/pkg_resources/__init__.py b/third-party/py/setuptools/pkg_resources/__init__.py
index d09e0b6f9af..c2d29fe358a 100644
--- a/third-party/py/setuptools/pkg_resources/__init__.py
+++ b/third-party/py/setuptools/pkg_resources/__init__.py
@@ -29,7 +29,6 @@
 import functools
 import pkgutil
 import token
-import symbol
 import operator
 import platform
 import collections
@@ -80,6 +79,7 @@
     importlib_machinery = None
 
 try:
+    import symbol
     import parser
 except ImportError:
     pass
